src/main/java/org/folio/linked/data/service/resource/ResourceMarcServiceImpl.java

```diff
@@ -230,8 +245,43 @@ private boolean isMonograph(String leader) {
     return isLanguageMaterial(leader.charAt(6)) && isMonographicComponentPartOrItem(leader.charAt(7));
   }
 
+  private Optional<ResponseEntity<Record>> getRecord(String inventoryId) {
+    try {
+      return Optional.of(srsClient.getFormattedSourceStorageInstanceRecordById(inventoryId));
+    } catch (FeignException.NotFound e) {
+      return Optional.empty();
+    }
+  }
+
+  private Optional<org.folio.ld.dictionary.model.Resource> getResource(String inventoryId) {
+    return getRecord(inventoryId)
+      .map(HttpEntity::getBody)
+      .map(Record::getParsedRecord)
+      .map(ParsedRecord::getContent)
+      .map(this::toJsonString)
+      .flatMap(marcBib2ldMapper::fromMarcJson);
+  }
+
+  private Long save(org.folio.ld.dictionary.model.Resource modelResource) {
```

Comment:
These changes will work. However, the class has become complex and less readable due to different `save**` and `create**` methods. I think we can split this class into two

1. ResourceMarcAuthorityService -> class for Marc authority records
2. ResourceMarcBibService -> class for Marc bib records.

I have created a DRAFT PR https://github.com/folio-org/mod-linked-data/pull/25 with this idea. Can you please take a look? I think we can implement this in scope of MODLD-535

@PBobylev  @AndreiBordak, What do you think?

Reply:
@pkjacob looks like a good idea